### ISSUE TITLE
How to use the imagenet local file to evaluate the models?

### ISSUE THREAD

**Issue by appleleaves** [larq/larq#168]
The examples in https://larq.dev/models/examples/ only show using the tensorflow imagenet dataset to test the model.
How can I evaluate the models using local ImageNet images?
How can I write the testloader?

**Comment by lgeiger** [comment]
> How can I evaluate the models using local ImageNet images?

There are many different ways to use a local version of ImageNet. If you have the images on disk you can replace the predict call in the following example with evaluate and pass in a batch of images with labels: https://larq.dev/models/examples/#classify-imagenet-classes-with-bi-real-net

`lqz.preprocess_input` takes either image as a `numpy` array or a Tensor, so it works independently from TensorFlow Datasets.

In general `larq-zoo` closely mirrors the Keras Applications API, so if you have an input pipeline that already works with Keras, it should work with `larq` and `larq-zoo` as well.

In generally I'd recommend to try out `tf.data` together with TensorFlow Datasets, which allows to construct very efficient data input pipelines. TensorFlow Datasets will also download all the images in a reproducible way. There is also a workaround to use it together with already downloaded ImageNet data: https://github.com/tensorflow/datasets/issues/538

> How can I write the testloader?

Could you elaborate?

**Comment by lgeiger** [comment]
@appleleaves Does this answer your question?
